trulens/nn/models/keras.py

```diff
@@ -242,10 +242,34 @@ def _get_layers(self, cut):
             (items if isinstance(items, DATA_CONTAINER_TYPE) else [items])
         ]
 
+        if cut.anchor not in ['in', 'out']:
+            return flat(
+                [
+                    layer.output[cut.anchor][0]
+                    if cut.anchor in layer.output else layer.output
+                    for layer in layers
+                ])
         return (
             flat([layer.input for layer in layers])
             if cut.anchor == 'in' else flat([layer.output for layer in layers]))
 
+    def _prepare_intervention_with_input(
+            self, model_args, intervention, doi_tensors):
+        input_tensors = self._get_layers(InputCut())
+        if not all(elem in doi_tensors for elem in input_tensors):
+
+            doi_tensors.extend(input_tensors)
+            if len(intervention[0]) != len(model_args[0]):
+                doi_factor = len(intervention[0]) / len(model_args[0])
```

Comment:
nit: avoid calling the len of the first element twice, just add a variable like `intervention_len` or something like that. same with model args. just so it's clearer what this check is doing.